src/client_creds.rs

```diff
@@ -23,22 +24,23 @@ use maybe_async::maybe_async;
 pub struct ClientCredsSpotify {
     pub config: Config,
     pub creds: Credentials,
-    pub token: Option<Token>,
+    pub token: RefCell<Option<Token>>,
     pub(in crate) http: HttpClient,
 }
 
 /// This client has access to the base methods.
+#[maybe_async(?Send)]
 impl BaseClient for ClientCredsSpotify {
     fn get_http(&self) -> &HttpClient {
         &self.http
     }
 
-    fn get_token(&self) -> Option<&Token> {
-        self.token.as_ref()
+    async fn get_token(&self) -> Ref<Option<Token>> {
+        self.token.borrow()
```

Comment:
By the way why does the client_credentials auth flow not re-authenticate?

Reply:
According to the [official document](https://developer.spotify.com/documentation/general/guides/authorization-guide/#authorization-flows), the client_credentials auth flow doesn't support access token refresh. If we want to implement re-authenticate for client credential auth flow, we should figure out another way to go.

Sorry for not mentioning this point.

Reply:
Since this doesn't require user interaction, could it perhaps be refreshed by just authenticating the usual way?